tests/contenttypes_tests/test_views.py

```diff
@@ -104,7 +104,55 @@ def test_shortcut_view_with_null_site_fk(self, get_model):
         obj = ModelWithNullFKToSite.objects.create(title='title')
         url = '/shortcut/%s/%s/' % (ContentType.objects.get_for_model(ModelWithNullFKToSite).id, obj.pk)
         response = self.client.get(url)
-        self.assertRedirects(response, '%s' % obj.get_absolute_url(), fetch_redirect_response=False)
+        correct_url = 'http://testserver{}'.format(obj.get_absolute_url())
+        self.assertRedirects(response, correct_url, fetch_redirect_response=False)
+
+    @mock.patch('django.apps.apps.get_model')
+    def test_shortcut_view_with_site_m2m(self, get_model):
+        """
+        When the object has a m2m relationship to sites,
+        redirect to the current site only if its attached to the object.
+        """
+        get_model.side_effect = lambda *args, **kwargs: MockSite if args[0] == 'sites.Site' else ModelWithM2MToSite
+
+        # get_current_site will lookup a Site object,
+        # so these need to match the same domains in the MockSite model.
+        Site.objects.bulk_create([
+            Site(pk=2, domain='example-2.com', name='example-2.com'),
+            Site(pk=3, domain='example-3.com', name='example-3.com'),
+        ])
+
```

Comment:
Less blank lines please. Some comments might take the place of blank lines. I'm not sure if the docstring does a good job at describing the purpose of the entire test. About some code, Carlton mentioned, "There's no explicit test covering it, but previously it was at least exercised but the test suite. It's now not run at all." -- is part of the test passing even before the fix? If so, we'd prefer to add that as a separate commit that precedes the fix to make it more clear what issue is fixed here and so that if the fix is reverted, for example, then that additional test coverage can stay.